cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -31,6 +31,8 @@
 #include "arrow/dataset/type_fwd.h"
 #include "arrow/dataset/visibility.h"
 #include "arrow/io/caching.h"
+#include "parquet/encryption/dataset_encryption_config.h"
```

Comment:
Should we add macros to these header files so the compiler won't complain when encryption is not enabled (e.g. `PARQUET_REQUIRE_ENCRYPTION` is OFF)? Or at least make sure they compile without encryption enabled.